### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by kevmoo**
A bunch of random cleanup

CC @devoncarew @sethladd @keertip

### CODE REVIEW

File: lib/src/html_generator.dart [hunk 1]
```diff
@@ -41,10 +41,10 @@ class Templates {
   TemplateRenderer topLevelPropertyTemplate;
   TemplateRenderer typeDefTemplate;
 
-  final Map<String, String> _partialTemplates = {};
+  final Map<String, String> _partialTemplates = <String, String>{};
```
Comment (sethladd) [inline lib/src/html_generator.dart]: why do we need the generics here? how does it help?  (do we ever do an `is` check on this object?)

**Comment by sethladd** [comment]
thanks! one comment to resolve. generally, we try not to add generics to the collections object itself unless we need to do an `is` check.

File: lib/src/html_generator.dart (hunk 1 above)
Comment (kevmoo) [inline lib/src/html_generator.dart]: Generics are reified. This allows you to get runtime exceptions in checked mode if the wrong kind of item is added.

**Comment by kevmoo** [comment]
I'm a big believer that having the type of the instance is *more* important than on the field.

**Pull request merged**